bql/planner/planner.go

```diff
@@ -788,6 +789,240 @@ func (p *queryPlan) String() string {
 	return b.String()
 }
 
+// constructPlan encapsulates the sequence of instructions that need to be
+// executed in order to satisfy the execution of a valid construct BQL statement.
+type constructPlan struct {
+	stm       *semantic.Statement
+	store     storage.Store
+	tracer    io.Writer
+	queryPlan *queryPlan
+}
+
+func (p *constructPlan) processReificationClause(rc *semantic.ReificationClause, tbl *table.Table, r table.Row) (*predicate.Predicate, *triple.Object, error) {
+	var err error
+	rprd, robj := rc.P, rc.O
+	if rprd == nil {
+		if tbl.HasBinding(rc.PBinding) {
+
+			// Try to bind the predicate.
+			v, ok := r[rc.PBinding]
+			if !ok {
+				return nil, nil, fmt.Errorf("row %+v misses binding %q", r, rc.PBinding)
+			}
+			if v.P == nil {
+				return nil, nil, fmt.Errorf("binding %q requires a predicate, got %+v instead", rc.PBinding, v)
+			}
+			rprd = v.P
+		} else if rc.PTemporal && rc.PAnchorBinding != "" {
+
+			// Try to bind the predicate anchor.
+			v, ok := r[rc.PAnchorBinding]
+			if !ok {
+				return nil, nil, fmt.Errorf("row %+v misses binding %q", r, rc.PAnchorBinding)
+			}
+			if v.T == nil {
+				return nil, nil, fmt.Errorf("binding %q requires a time, got %+v instead", rc.PAnchorBinding, v)
+			}
+			rprd, err = predicate.NewTemporal(rc.PID, *v.T)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+	}
+	if robj == nil {
+		if tbl.HasBinding(rc.OBinding) {
+
+			// Try to bind the object
+			v, ok := r[rc.OBinding]
+			if !ok {
+				return nil, nil, fmt.Errorf("row %+v misses binding %q", r, rc.OBinding)
+			}
+			co, err := cellToObject(v)
+			if err != nil {
+				return nil, nil, err
+			}
+			robj = co
+		} else if rc.OTemporal && rc.OAnchorBinding != "" {
+
+			// Try to bind the object anchor.
+			v, ok := r[rc.OAnchorBinding]
+			if !ok {
+				return nil, nil, fmt.Errorf("row %+v misses binding %q", r, rc.OAnchorBinding)
+			}
+			if v.T == nil {
+				return nil, nil, fmt.Errorf("binding %q requires a time, got %+v instead", rc.OAnchorBinding, v)
+			}
+			rop, err := predicate.NewTemporal(rc.OID, *v.T)
+			if err != nil {
+				return nil, nil, err
+			}
+			robj = triple.NewPredicateObject(rop)
+		}
+
+	}
+	return rprd, robj, nil
+}
+
+func (p *constructPlan) Execute(ctx context.Context) (*table.Table, error) {
+	tbl, err := p.queryPlan.Execute(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	tripChan := make(chan *triple.Triple, 10)
+	done := make(chan bool)
+
+	go func() {
+		ts := []*triple.Triple{}
+		updateFunc := func(g storage.Graph, d []*triple.Triple) error {
+			trace(p.tracer, func() []string {
+				return []string{"Inserting triples to graph \"" + g.ID(ctx) + "\""}
+			})
+			return g.AddTriples(ctx, d)
+		}
+		for elem := range tripChan {
+			ts = append(ts, elem)
+			if len(ts) >= 10 {
```

Comment:
Done. Have plumbed BulkTripleOpSize through to here and ensured that all tests pass and bw compiles and runs.